### DISCUSSION TITLE
Ignoring some code ranges or queries for Ray.

### DISCUSSION THREAD

**Discussion by mertasan** [spatie/ray#489] [Ideas]
Sometimes I need to enable the `"send_queries_to_ray"` setting. But some queries are unnecessary as in the screenshot below. It's taking up the screen and I don't want to see these queries. I have to press `Cmd+L` every time. I know I can use `ray()->clearAll()` but sometimes previous queries are not supposed to be cleared.

![screenshot 40](https://user-images.githubusercontent.com/13007665/123526444-9b36a480-d6e0-11eb-8552-1d230e0f1052.png)

I have two suggestions.

1: 
```php
ray()->off();
// Line ranges where Ray should not print any events.
ray()->on();
```

2:
```php
// ray.php
'ignore' => [
  '\Illuminate\Session\DatabaseSessionHandler::performUpdate', // or something like that :)
]
```

**Comment by freekmurze** [comment]
Have you tried using `ray()->disable()`?

**Reply by mertasan** [reply]
I guess if I do as you say I will have to interfere with Laravel's root files. Or I need to write a macro.

It would be nice if there was a way to set this from the ray.php file.

For example, I enable the `send_queries_to_ray` setting.
I am running a unit test for a laravel plugin.
I want to see the SQL queries generated in the plugin. However, dozens of SQL queries generated for unit testing are printed to the screen.

![screenshot 45](https://user-images.githubusercontent.com/13007665/123720340-8434b580-d88c-11eb-8d1f-024eee13d530.png)

Isn't it more convenient to set this from ray.php?

I think what I want is very useful, I know it is difficult to decide what is the best thing to do.

Some methods in the PhpStorm [metadata](https://www.jetbrains.com/help/phpstorm/ide-advanced-metadata.html) can serve as an inspiration for this.

It would be nice if there was a smart filtering. I can't speak English well enough. That's why I can't explain my thoughts properly. I hope I was able to explain what I mean. :) thanks

**Reply by mertasan** [reply]
@freekmurze The content of the screenshot above is created from the codes below.

Therefore, it is necessary to interfere with other related files to disable it.
  
```
// TestCase.php
public function migrate (array $options = []): void
{
    ray()->disable();
    $this->artisan('migrate', $options)->run();
    ray()->enable();
}
```
```
// Pest.php
function migrate (array $options = [])
{
    return test()->migrate($options);
}
```

```
// Test.php
it('test', function(){
    ...
    migrate();
    ...
});
```

![screenshot 45](https://user-images.githubusercontent.com/13007665/123722892-e93eda00-d891-11eb-9b64-dc0abe866bea.png)
